client/src/pages/EditProfile/EditProfileForm/EditProfileForm.tsx

```diff
@@ -0,0 +1,96 @@
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import Box from '@material-ui/core/Box';
+import { Formik, FormikHelpers } from 'formik';
+import * as Yup from 'yup';
+import Typography from '@material-ui/core/Typography';
+import useStyles from './useStyles';
+import { CircularProgress } from '@material-ui/core';
+import { User } from '../../../interface/User';
+interface Props {
+  handleSubmit: (
+    {
+      newUsername,
+      newEmail,
+    }: {
+      newUsername: string;
+      newEmail: string;
+    },
+    {
+      setStatus,
+      setSubmitting,
+    }: FormikHelpers<{
+      newUsername: string;
+      newEmail: string;
+    }>,
+  ) => void;
+  loggedInUser: User;
+}
+
+const EditProfileForm = ({ handleSubmit, loggedInUser }: Props): JSX.Element => {
+  const classes = useStyles();
+
+  return (
+    <Formik
+      initialValues={{
+        newEmail: loggedInUser.email,
+        newUsername: loggedInUser.username,
+      }}
+      validationSchema={Yup.object().shape({
+        newUsername: Yup.string().required('Username is required').max(40, 'Username is too long'),
+        newEmail: Yup.string().required('Email is required').email('Email is not valid'),
+      })}
+      onSubmit={handleSubmit}
```

Comment:
I generally recommend defining these objects before render/return (e.g. outside of the JSX) to promote clarity and readability 👍🏼 

Material UI is particularly good at this - check out how clear the UI of [`Button`](https://github.com/mui-org/material-ui/blob/c545ccab7edfdf4a44d4ec2f4bf10ebc7fd00259/packages/material-ui/src/Button/Button.js#L294) and [`TextField`](https://github.com/mui-org/material-ui/blob/c545ccab7edfdf4a44d4ec2f4bf10ebc7fd00259/packages/material-ui/src/TextField/TextField.js#L160) components are despite their complex implementations!

Reply:
I think I understand what you mean but can you clarify what you mean by how clear the UI is?